src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -63,6 +63,49 @@ public void Kill()
             }
         }
 
+        /// <summary>Stops the associated process immediately.</summary>
+        private void Stop()
+        {
+            EnsureState(State.HaveNonExitedId);
+            if (Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGSTOP) != 0)
+            {
+                throw new Win32Exception(); // to match behavior of Kill()
+            }
+        }
+
+        /// <summary>
+        /// Makes a best-effort attempt to kill all descendant (child, grandchild, etc.) processes.
+        /// </summary>
+        private void KillTree()
+        {
+            try
+            {
+                // Stop but don't kill the process. Keeps additional children from being started but leaves the process around 
+                // so that its children can be enumerated.
+                Stop();
+            }
+            catch
+            {
+                // Making a best attempt. If it fails (perhaps because the process is already dead), give up.
+                return;
+            }
+
+            IReadOnlyList<Process> children = GetChildProcesses();
+
+            try
+            {
+                // Since the process's children have been enumerated, it can now be terminated.
+                Kill();
+            }
+            catch
+            {
+                // Making a best attempt. If it fails (perhaps because the process is already dead), give up.
+                return;
```

Comment:
If you fail to terminate the process here, why not continue to try to terminate the children? Perhaps eg the parent has exited of its own accorded just moments ago.

Reply:
Same goes for the other return statement above

Reply:
Good idea. Change made. Thanks!